### DISCUSSION TITLE
Keyboard shortcuts don't behave as expected

### DISCUSSION THREAD

**Discussion by codeanpeace** [zoonk/uneebee#173] [Bugs & Issues]
Specifically, tabbing skips the answers and requires clicking. It'd be nice to see tab and arrow key support. It'd be really cool if there was a keyboard friendly way of selecting answers ala Typeform. Also, the confirm button seems unnecessary and just slows down the process in my book.

Otherwise, nice project and best of luck with it!

**Comment by ceolinwill** [comment]
> Specifically, tabbing skips the answers and requires clicking. It'd be nice to see tab and arrow key support. It'd be really cool there was a keyboard friendly way of selecting answers ala Typeform.

I agree. I've created an issue for this: https://github.com/zoonk/uneebee/issues/118

> the confirm button seems unnecessary and just slows down the process in my book.

I see how this can be annoying in certain cases. The confirm button is a way to avoid selecting the wrong answer by accident. If we remove it, then we'd have two options:

**1. Automatically confirm the selected answer**

This means users could potentially click/select them by mistake and there's no way to change it.

**2. Clicking on "next" confirms the selection**

If clicking on `next` confirms the selected answer, then it should navigate to the next step and display the feedback message there, which means users would have to close it and have a second click anyways.

---

I'm actually copying this workflow from Duolingo. You select an option, click on `check`, it displays the feedback message, then you click on `next step` to navigate to the next question.

**Reply by codeanpeace** [reply]
Yeah, that's fair! I guess it's just more noticeable on a desktop environment – especially without typeform-esque keyboard shortcuts – vs a mobile touchscreen environment. After clicking/tabbing/arrowing to select an answer, it'd be nice if hitting enter results in a confirm. Even nicer would be a Typeform or Vimium esque way of "clicking" via keyboard shortcuts, but it's arguably a more power user approach.

**Reply by ceolinwill** [reply]
Yeah, I think it will be better after I implement keyboard shortcuts. I'll try to include them for v1. I just had some health issues over the past month that prevented me from working on this but I'm slowly getting back to it. :)

> Even nicer would be a Typeform or Vimium esque way of "clicking" via keyboard shortcuts

Totally. I think we can have something like pressing `1` to select the first answer, `2` to select the second one and so, then `enter` to confirm and `enter` again to move to the next step.
